src/components/stone-handler.ts

```diff
@@ -292,4 +315,86 @@ export default class StoneHandler extends EventManager {
     // to play the audio parrallely.
     this.correctStoneAudio.play();
   }
+
+  resetStonePosition(
+    width,
+    pickedStone,
+    pickedStoneObject
+  ) {
+    const stone = pickedStone;
+    const stoneObj = pickedStoneObject;
+    //Resets the previous stone letter to its original position.
+    if (
+        stone &&
+        stoneObj &&
+        stone.text &&
+        typeof stoneObj.origx === "number" &&
+        typeof stoneObj.origy === "number"
+    ) {
+      const xLimit = 50;
+      const halfWidth = width / 2;
+
+      stone.x = stone.text.length <= 3 &&
+        stoneObj.origx < xLimit &&
+        stoneObj.origx < halfWidth
+          ? stoneObj.origx + 25
+          : stoneObj.origx;
+      stone.y = stoneObj.origy;
+    }
+
+    return stone;
+  }
+
+  private computeCursorDistance(posX, posY, sc):number {
+    return Math.sqrt((posX - sc.x) ** 2 + (posY - sc.y) ** 2);
+  }
+
+  handlePickStoneUp(posX, posY) {
+    let stoneLetter = null;
+    let ctr = 0;
+    for (let sc of this.foilStones) {
+      const distance = this.computeCursorDistance(posX, posY, sc);
+      if (distance <= 40) {
+          stoneLetter = sc;
+          /* Adds a unique identifier to tell which letter is which in case there are two or more of the same letter.*/
+          stoneLetter['foilStoneIndex'] = ctr;
+          break;
+      }
+      ctr++;
+    };
+
+    return stoneLetter;
+  }
+
+  handleHoveringToAnotherStone(
+    posX,
+    posY,
+    shouldGroupLetter
+  ) {
+    /* Handle hovering of stones for word puzzle multi-letter select.*/
+    let stoneLetter = null;
+    let ctr = 0;
+    for (let sc of this.foilStones) {
+      const distance = this.computeCursorDistance(posX, posY, sc);
+
+      if (distance <= 40 && shouldGroupLetter(sc.text, ctr)) {
+        stoneLetter = sc;
+        /* Adds a unique identifier to tell which letter is which in case there are two or more of the same letter.*/
+        stoneLetter['foilStoneIndex'] = ctr;
+        break;
+      }
+      ctr++;
+    };
+
+    return stoneLetter;
+  }
+
+  handleMovingStoneLetter(draggingStone, posX, posY) {
+    const updatedStoneCoordinates = draggingStone;
```

Comment:
In handleMovingStoneLetter, the variable const updatedStoneCoordinates = draggingStone; can be avoided since you’re mutating the object. Directly use draggingStone